gateway-service/src/main/java/org/zowe/apiml/gateway/zaas/ZaasController.java

```diff
@@ -69,4 +78,36 @@ public ResponseEntity<Object> getPassTicket(@RequestBody TicketRequest ticketReq
             .status(HttpStatus.OK)
             .body(new TicketResponse(null, authSource.getUserId(), applicationName, ticket));
     }
+
+    @PostMapping(path = "zosmf", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Provides z/OSMF JWT or LTPA token for authenticated user.")
+    @ResponseBody
+    public ResponseEntity<Object> getZosmfToken(@RequestAttribute(AUTH_SOURCE_ATTR) AuthSource.Parsed authSource, HttpServletRequest request) {
+
+        Optional<AuthSource> rawAuthSource = authSourceService.getAuthSourceFromRequest(request);
```

Comment:
It will be probably better to store this value as an attribute in the request (not only parsed value) - it is redundant. But I guess it is out of the scope...